python_modules/dagster/dagster/core/asset_defs/materialize.py

```diff
@@ -40,3 +42,43 @@ def materialize(
         assets=assets_defs,
         source_assets=source_assets,
     ).execute_in_process(run_config=run_config, instance=instance)
+
+
+def materialize_in_process(
```

Comment:
Even though I think I might have been the one who proposed this name, I'm not 100% sure it's the right one. In the context of `execute_in_process`, "in-process" mostly means the in-process executor. But `materialize` also uses the in-process executor. The main difference here is that we use the in-memory IO manager. So maybe we should call this `materialize_to_memory`?